src/main/java/tz/go/moh/him/nhcr/mediator/orchestrator/BaseOrchestrator.java

```diff
@@ -14,13 +19,26 @@
      */
     protected final MediatorConfig config;
 
+    /**
+     * Represents an Error Messages Definition Resource Object defined in <a href="file:../resources/error-messages.json">/resources/error-messages.json</a>.
+     */
+    protected JSONObject errorMessageResource;
+
     /**
      * Initializes a new instance of the {@link DefaultOrchestrator} class.
      *
      * @param config The configuration.
      */
     protected BaseOrchestrator(MediatorConfig config) {
         this.config = config;
+        InputStream stream = getClass().getClassLoader().getResourceAsStream("error-messages.json");
+        try {
+            if (stream != null) {
+                errorMessageResource = new JSONObject(IOUtils.toString(stream));
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
We should log the exception using the standard logger for the mediators, similar to the orchestrators.

`protected final LoggingAdapter log = Logging.getLogger(getContext().system(), this);`

Reply:
We also might want to prevent the mediator from starting if we cannot read the error messages file, because there is other code in the mediator that depends on the existence of the error messages resource.